prov/sockets/src/sock_progress.c

```diff
@@ -1189,9 +1193,20 @@ static int sock_pe_progress_buffered_rx(struct sock_rx_ctx *rx_ctx)
 	    dlist_empty(&rx_ctx->rx_buffered_list))
 		return 0;
 
-	for (entry = rx_ctx->rx_buffered_list.next;
-	     entry != &rx_ctx->rx_buffered_list;) {
-
+        if (depth < 0) {
+            /* check entire buffered list */
+            entry = &rx_ctx->rx_buffered_list;
+        } else {
+            /* continue where last time left off */
+            entry = rx_ctx->progress_start;
+        }
+        if (entry == &rx_ctx->rx_buffered_list) {
+            entry = entry->next;
+        }
+        for (int i_prog = 0; depth < 0 || i_prog < SOCK_EP_MAX_PROGRESS_CNT; i_prog++) {
+                if (entry == &rx_ctx->rx_buffered_list) {
+                    break;
+                }
```

Comment:
And if statement that breaks as the first code inside a for loop should just move the check into the for loop itself.